api-get-policy.go

```diff
@@ -35,7 +35,8 @@ func (c Client) GetBucketPolicy(bucketName, objectPrefix string) (bucketPolicy p
 		return policy.BucketPolicyNone, err
 	}
 	policyInfo, err := c.getBucketPolicy(bucketName)
-	if err != nil {
+	errResponse := ToErrorResponse(err)
+	if err != nil && errResponse.Code != "NoSuchBucketPolicy" {
```

Comment:
It fixes #664 because Gateway uses GetBucketPolicy() Core API directly, which would return NoSuchBucketPolicy i.e error when no bucket policy exists. This makes it easier for Gateway to have a pass-through error back to the client.